cmd/neofs-node/policy.go

```diff
@@ -2,26 +2,113 @@ package main
 
 import (
 	"fmt"
+	"sync"
 
+	"github.com/hashicorp/golang-lru/v2/simplelru"
 	"github.com/nspcc-dev/neofs-node/pkg/core/container"
 	"github.com/nspcc-dev/neofs-node/pkg/core/netmap"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
 	netmapsdk "github.com/nspcc-dev/neofs-sdk-go/netmap"
 )
 
+// storagePolicyRes structures persistent storage policy application result for
+// particular container and network map incl. error.
+type storagePolicyRes struct {
+	nodeSets [][]netmapsdk.NodeInfo
+	err      error
+}
+
+// containerNodesAtEpoch is a thread-safe LRU cache mapping containers into
+// storage policy application results for particular network map.
+type containerNodesAtEpoch struct {
+	mtx sync.RWMutex
+	lru simplelru.LRUCache[cid.ID, storagePolicyRes]
```

Comment:
why have you decided to use the simple version of it?

Reply:
Looks strange given that we have an `mtx` here anyway dealing with concurrency.

Why not a `[cid, epoch] -> [nodes, error]`, btw?

Reply:
> why have you decided to use the simple version of it?

to not make several calculations simultaneously. Sync cache does not prevent this

Reply:
> Why not a [cid, epoch] -> [nodes, error], btw?

to not block current epoch cache due to the acquired previous one